fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/ArrayDistinctValue.java

```diff
@@ -0,0 +1,171 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.query.plan.cascades.values;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.annotation.SpotBugsSuppressWarnings;
+import com.apple.foundationdb.record.EvaluationContext;
+import com.apple.foundationdb.record.ObjectPlanHash;
+import com.apple.foundationdb.record.PlanDeserializer;
+import com.apple.foundationdb.record.PlanHashable;
+import com.apple.foundationdb.record.PlanSerializationContext;
+import com.apple.foundationdb.record.planprotos.PArrayDistinctValue;
+import com.apple.foundationdb.record.planprotos.PValue;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStoreBase;
+import com.apple.foundationdb.record.query.plan.cascades.AliasMap;
+import com.apple.foundationdb.record.query.plan.cascades.typing.Type;
+import com.apple.foundationdb.record.query.plan.explain.ExplainTokens;
+import com.apple.foundationdb.record.query.plan.explain.ExplainTokensWithPrecedence;
+import com.google.auto.service.AutoService;
+import com.google.common.base.Verify;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.Message;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+
+/**
+ * A value that returns a copy of its child array {@link Value} with all duplicate elements removed.
+ */
+@API(API.Status.EXPERIMENTAL)
+public class ArrayDistinctValue extends AbstractValue implements ValueWithChild {
+    private static final ObjectPlanHash BASE_HASH = new ObjectPlanHash("Array-Distinct-Value");
+
+    @Nonnull
+    private final Value childValue;
+    @Nonnull
+    private final Type resultType;
+
+    public ArrayDistinctValue(@Nonnull final Value childValue) {
+        final var innerResultType = Objects.requireNonNull(childValue.getResultType());
+        Verify.verify(innerResultType.isArray());
+        this.childValue = childValue;
+        this.resultType = innerResultType;
+    }
+
+    @Nonnull
+    @Override
+    public List<? extends Value> computeChildren() {
+        return ImmutableList.of(childValue);
+    }
+
+    @Nonnull
+    @Override
+    public Value getChild() {
+        return childValue;
+    }
+
+    @Nonnull
+    @Override
+    public ValueWithChild withNewChild(@Nonnull final Value rebasedChild) {
+        return new ArrayDistinctValue(rebasedChild);
```

Comment:
Can you do an `== ` comparison here and not return a new instance if that is the same?